src/lib/logger.js

```diff
@@ -6,13 +6,18 @@ const chalk = require('chalk');
 module.exports = {
   log(...args) {
     const [format, ...rest] = args;
-    console.log(`${chalk.grey('[Semantic release]:')} ${format}`, ...rest.map(arg => chalk.magenta(arg)));
+    console.log(
+      `${chalk.grey('[Semantic release]:')}${typeof format === 'string' ? format : ''}`,
```

Comment:
`console.log()` will check if the first arg contains any format sequences, not just if it's a string. I.e. these two act different:
```
console.log('an error occured: ', err)
console.log('an error occured: %s', err)
```

Reply:
That's expected I think.
`console.log('an error occured: ', err)` will log `err` as an object (with stacktrace etc)
`console.log('an error occured: %s', err)` will log only the equivalent of `err.toString()`

If a user use `%s` in their format it means they want to be logged a string.

Can you give me an example that won't work as expected ?

Reply:
also here's one idea: why not simply wrap all format sequences with chalk? I.e.
```ts
if (typeof format === 'string') {
  format.replace(/%[^%]/g, seq => chalk.magenta(seq))
}
```

Reply:
> console.log('an error occured: ', err) will log err as an object (with stacktrace etc)

Are you sure? As I understand the code this will wrap `err` in `chalk.magenta()` because the first argument is a string and it doesn't check whether it actually contains a format sequence like `console` does.

Reply:
```js
logger.log('this is an object', {a: 'a'});
// [Semantic release]: this is an object { a: 'a' }
```
```js
const error = new Error('err message');
error.property = 'propValue';
logger.error('this is an error', error);

//[Semantic release]: this is an error { Error: err message
//    at module.exports (/Users/vanduynslager/git/semantic-release/src/cli.js:57:21)
//    at <anonymous> property: 'propValue' }
```

Reply:
> also here's one idea: why not simply wrap all format sequences with chalk? I.e.

That seems a good idea. Let me investigate a bit more (tomorrow though).

Reply:
I implemented the `format.replace(/%[^%]/g, seq => chalk.magenta(seq))`. Thanks @felixfbecker !